### DISCUSSION TITLE
textlayout::TextStyle How to add underline and line-through at the same time

### DISCUSSION THREAD

**Discussion by StringKe** [rust-skia/rust-skia#452] [Q&A] [answered]
`skia_safe::textlayout::TextStyle` How to add underline and line-through at the same time

**Comment by pragmatrix** [comment, marked as answer]
Hi, `TextDecoration` is defined through a [`bitflags!` macro](https://docs.rs/bitflags/1.2.1/bitflags/), so it should be possible to simply `|` "or" them.

**Reply by StringKe** [reply]
canvaskit is a skia module, i find this [wasm demo](https://jsfiddle.skia.org/canvaskit/41531075862331fe26d39dfcb0d923c3ea7ba8a90ccdfc15a6dcb1127b7c9ebf) the 46 line decoration input 0~8 toggle multiple effects. i find source is [here](https://github.com/google/skia/blob/master/modules/canvaskit/paragraph_bindings.cpp#L151)  and [here](https://github.com/google/skia/blob/master/modules/skparagraph/include/TextStyle.h#L38) is a constant but he can pass parameters.

I don't quite understand, can you answer it?

**Reply by pragmatrix** [reply]
This should work:

```
    let mut ts = TextStyle::new();
    ts.decoration_mut().ty = TextDecoration::UNDERLINE | TextDecoration::LINE_THROUGH;
```

**Reply by StringKe** [reply]
thx, it is work
